docs/references/execution-errors.mdx

```diff
@@ -386,14 +386,346 @@ There is a [limit][limits] on the total size in bytes of the [code section](http
+
+To fix this error, consider [adding the required cycles][topping-up] to the canister or otherwise reducing the canister's memory usage.
+
+
+### Reserved cycles limit exceeded in memory allocation
+There was an attempt to increase the canisters memory allocation that would require reserving more cycles than allowed by the canister's reserved cycles limit.
+
+An example of this error is:
+```
+  Cannot increase memory allocation to 50000000000 due to its reserved cycles limit. The current limit (1000000000000) would be exceeded by 301000000.
+```
+
+When subnets start to run low on memory, using up more memory requires reserving cycles to pay for the future use of that memory. Canisters have a setting that limits the number of cycles that they will reserve and this error indicates that a canister failed to increase its [memory allocation][memory-allocation-setting] because doing so would cause the number of reserved cycles to exceed that limit.
+
+To diagnose this error use the [`dfx canister status`][dfx-canister-status] command or the [`canister_status` API][canister-status-api] to check the canister's current reserved cycles limit. The subnet memory usage can also be seen on the [ICP dashboard][dashboard-subnets].
+
+To fix this error, consider increasing the canister's reserved cycles limit, or moving to a subnet with lower memory usage.
+
+
+### Wasm chunk store error
+There was an error while executing a method to manipulate the canister's [Wasm chunk store](docs/current/references/ic-interface-spec/#ic-upload_chunk) (e.g. `upload_chunk`, `clear_chunk_store`, `install_chunked_code`, etc.).
+
+An example of this error is:
+```
+  Error from Wasm chunk store: Cannot upload chunk. At least 10000000 additional cycles are required.
+```
+
+To fix this error, consider [topping up][topping-up] the canister if it doesn't have enough cycles, or using the [`stored_chunks` API](docs/current/references/ic-interface-spec/#ic-stored_chunks) to see which chunks have been uploaded if you get errors about mismatching or missing hashes.
+
+
+### Canister snapshot not found
+A canister snapshot operation was performed with a snapshot ID that doesn't exist.
+
+An example of this error is:
+```
+  Could not find the snapshot ID 125-xxx-xxx for canister xxx-xxx.
+```
+
+To fix this error, consider using the `list_canister_snapshot` API to see which snapshots exist on your canister.
+
+
+### Canister heap delta rate limited
+The canister has been rate limited which prevents it from taking or loading a snapshot.
+
+An example of this error is:
+```
+  Canister xxx-xxx is heap delta rate limited: current delta debit is 1000005000, but limit is 1000000000.
+```
+
+Canisters are limited by how much they can write within a given time frame. This error indicates that the canister has recently written a large amount of data which means it cannot currently take or load a snapshot because those operations also generate writes to the blockchain state.
```

Comment:
```suggestion
Canisters are limited by how much they can write within a given time frame. This error indicates that the canister has recently written a large amount of data and it cannot currently take or load a snapshot because those operations also generate writes to the blockchain state.
```